tests/e2e/default/tasks/tasks-breadcrumbs.wdio-spec.js

```diff
@@ -0,0 +1,156 @@
+  patient_id: 'patient1',
+  parent: { _id: clinic._id, parent: { _id: health_center._id, parent: { _id: district_hospital._id }}},
+  reported_date: new Date().getTime(),
+});
+const patient2 = personFactory.build({
+  _id: 'patient2',
+  name: 'patient2',
+  type: 'person',
+  patient_id: 'patient2',
+  parent: { _id: health_center._id, parent: { _id: district_hospital._id }},
+  reported_date: new Date().getTime(),
+});
+
+const getTasksInfos = async (tasks) => {
+  const infos = [];
+  for (const task of tasks) {
+    infos.push(await tasksPage.getTaskInfo(task));
+  }
+  return infos;
+};
+
+describe('Tasks tab breadcrumbs', () => {
+  before(async () => {
+    await utils.saveDocs([ ...places.values(), contact, contact2, patient, patient2 ]);
+    await utils.createUsers([ chw, supervisor ]);
+    await sentinelUtils.waitForSentinel();
+
+    await chtConfUtils.initializeConfigDir();
+
+    const formsPath = path.join(__dirname, 'forms');
+    await chtConfUtils.compileAndUploadAppForms(formsPath);
+
+    const tasksFilePath = path.join(__dirname, 'tasks-breadcrumbs-config.js');
+    const { tasks } = await chtConfUtils.compileNoolsConfig({ tasks: tasksFilePath });
+    await utils.updateSettings({ tasks }, 'api');
+  });
+
+  describe('for chw', () => {
+    before(async () => {
+      await loginPage.login(chw);
+    });
+
+    after(async () => {
+      await browser.deleteCookies();
+      await browser.refresh();
+    });
+
+    it('should display correct tasks with breadcrumbs for chw', async () => {
+      await tasksPage.goToTasksTab();
+      const infos = await getTasksInfos(await tasksPage.getTasks());
+
+      expect(infos).to.have.deep.members([
+        {
+          contactName: 'patient1',
+          formTitle: 'person_create',
+          lineage: clinic.name,
+          dueDateText: 'Due today',
+          overdue: true
+        },
+        {
+          contactName: 'patient2',
+          formTitle: 'person_create',
+          lineage: '',
+          dueDateText: 'Due today',
+          overdue: true
+        },
+      ]);
+    });
+  });
+
+  describe('for supervisor', () => {
+    before(async () => {
+      await loginPage.login(supervisor);
+    });
+
+    after(async () => {
+      await browser.deleteCookies();
+      await browser.refresh();
+    });
+
+    it('should display correct tasks with breadcrumbs for supervisor', async () => {
+      await tasksPage.goToTasksTab();
+      const infos = await getTasksInfos(await tasksPage.getTasks());
+
+      expect(infos).to.have.deep.members([
+        {
+          contactName: 'patient1',
+          formTitle: 'person_create',
+          lineage: clinic.name+health_center.name,
+          dueDateText: 'Due today',
+          overdue: true
+        },
+        {
+          contactName: 'patient2',
+          formTitle: 'person_create',
+          lineage: health_center.name,
+          dueDateText: 'Due today',
+          overdue: true
+        },
+      ]);
+    });
```

Comment:
I would love to see one more scenario testing with an online user `describe('for online user', () => {...` `it('should display correct tasks with breadcrumbs for online user', async ()...`

Reply:
I tried, but with the default config as I understand it, and as it usually the case the  `"can_view_tasks"` is only activated for chw and supervisor. Online users as program officers don't usually do tasks. Usually for breadcrumbs I test offline and online user cases, but here I added supervisor even though the logic is pretty much the same for both of them, when it comes to how we display breadcrumbs.

Reply:
Perfect! thanks for clarifying.